02-wrangling.Rmd

```diff
@@ -50,23 +65,45 @@ and the columns are the variables:
-For example, the data frame below has 3 elements that are vectors whose names are `state`, `year` and `population`.
+From a computer programming perspective, in R, a data frame is a special subtype of a list object whose elements (columns) are *vectors*. For example, the data frame in Figure \@ref(fig:02-vectors) has three vectors whose names are `region`, `year` and `population`.
 
 ```{r 02-vectors, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = "Data frame with 3 vectors", fig.retina = 2, out.width = "650"}
 knitr::include_graphics("img/vectors.jpeg")
 ```
 
 ### What is a vector?
 
-In R, vectors are objects that can contain 1 or more elements. The vector elements are ordered, and they must all be of the same type. Common 
-example types of vectors are character (e.g., letter or words), numeric (whole numbers and fractions) and logical (e.g., `TRUE` or `FALSE`). In 
-the vector shown below, the elements are of numeric type:
+In R, **vectors** are objects that can contain one or more elements. The vector elements are ordered, and they must all be of the same *data type*. R has several different basic data types, as shown in table \@ref(tab:datatype-table).
+
+In the vector shown in Figure \@ref(fig:02-vector), the elements are of integer type:
 
 ```{r 02-vector, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = "Example of a numeric type vector", fig.retina = 2, out.width = "600"}
 knitr::include_graphics("img/vector.jpeg")
 ```
 
+
+You can create vectors in R using the concatenate `c()` function. 
+To create the vector `year` in Figure \@ref(fig:02-vector) we write:
+```{r}
+year <- c(2015, 2013, 2011, 2016, 2018)
+year
+```
+
+Table: (\#tab:datatype-table) Basic data types in R
+
+| Data type | Description | Example |
+|-------|------------------|-----------|
+| character | Letters or numbers surrounded by quotes |  "1" , "Hello world!" |
+|  double  | Numbers with decimals values         | 1.2333 | 
+|  integer | Numbers that do not contain decimals | 1L, 20L (where "L" tells R to store as an integer) | 
+| logical |  A value of true and false | `TRUE`, `FALSE` | 
+| factor  | Used to represent data with a limited number of values | a `size` variable with levels "large", "medium" and "small" | 
+
+>  "Numeric" is an umbrella term that includes integer and double data types.
+> Most of the time when working with numbers in R they will be double types. For instance, a double data type is the default when you create a vector of numbers using `c()`, and when you read in whole numbers via `read_csv`.
+> There are other basic data types in R, such as *raw* and *complex*, which we won't cover in this textbook. 
+> We will go into more details about factors in chapter \@ref(viz).
+
 ### How are vectors different from a list?
 
 Lists are also objects in R that have multiple elements. Vectors and lists differ by the requirement of element type consistency. All elements 
```

Comment:
OK, from Jenny: good for us to call these über-vectors, so let's do that!

Reply:
Also, she has this great way of defining atomic vectors and I think we should do it how she does. As in, we actually call them atomic vectors when we introduce them, and define the as Jenny does (pasted below) and then tell them that we will call them vectors going forward. 

From Jenny:

Since then (or maybe I’ve said/written it there too), I feel like I’ve gotten a good way to explain atomic vectors. We’re saying what the atoms of these things are, i.e. the individual units from which they are built. Atomic vectors are collections built up from the most fundamental data types R has.